support/k8s/k8s-workload-registrar/mode-crd/api/spiffeid/v1beta1/spiffeid_types.go

```diff
@@ -47,10 +47,11 @@ type Selector struct {
 
 // SpiffeIDSpec defines the desired state of SpiffeID
 type SpiffeIDSpec struct {
-	ParentId string   `json:"parentId"`
-	SpiffeId string   `json:"spiffeId"`
-	Selector Selector `json:"selector"`
-	DnsNames []string `json:"dnsNames,omitempty"`
+	ParentId      string   `json:"parentId"`
+	SpiffeId      string   `json:"spiffeId"`
+	Selector      Selector `json:"selector"`
+	DnsNames      []string `json:"dnsNames,omitempty"`
+	FederatesWith []string `json:"federatesWith,omitempty"`
```

Comment:
I think this will require updating the api version. we are planning to do this anyways, so perhaps we can lump this in so we don't have to update too many times?

Reply:
I'm not planning to use the CRD mode, is it ok if we push this through without updating the API version and it gets picked up later?

Reply:
ok with me